tools/cassandra_delete_range/main.go

```diff
@@ -424,30 +477,43 @@ func prepareDeleteQueries(cluster *gocql.ClusterConfig, fromLedgerIdx uint64, qu
 
 				sessionCreationWaitGroup.Done()
 				sessionCreationWaitGroup.Wait()
-				preparedQueryString := fmt.Sprintf(queryTemplate, "?", "?")
-				preparedQuery := session.Query(preparedQueryString)
+				preparedQuery := session.Query(q)
 
 				for r := range rangesChannel {
 					preparedQuery.Bind(r.StartRange, r.EndRange)
-					iter := preparedQuery.Iter()
 
-					var key []byte
-					var seq uint64
+					var pageState []byte
 					var rowsRetrieved uint64
 
-					for iter.Scan(&key, &seq) {
-						rowsRetrieved++
+					for {
+						iter := preparedQuery.PageSize(*clusterPageSize).PageState(pageState).Iter()
+						nextPageState := iter.PageState()
+						scanner := iter.Scanner()
+
+						for scanner.Next() {
```

Comment:
Perhaps one way to improve the performance of the tool is to also move this internal loop as a separate goroutine. This would allow you to read the table concurrently. Though, that being said, it depends on whether `preparedQuery` supports creating multiple iterators at the same time.